cfripper/rules/wildcard_resource_rule.py

```diff
@@ -47,20 +46,24 @@ class WildcardResourceRule(ResourceSpecificRule):
     REASON_ALL_ACTIONS_WITHOUT_POLICY_NAME = '"{}" is using a wildcard resource allowing all actions'
 
     def resource_invoke(self, resource: Resource, logical_id: str, extras: Optional[Dict] = None) -> Result:
+        """
+        Checks each policy of a given resource.
+        If it's an IAMRole, it will check its AssumeRolePolicyDocument as well.
+        There are some cases where GenericResource contains a property called PolicyDocument that can be a str and
+        therefore, it's not being retrieved in the initial for loop.
+        For those cases, we run another check transforming the str to a PolicyDocument.
+        """
         result = Result()
 
         for policy in resource.policy_documents:
             self._check_policy_document(result, logical_id, policy.policy_document, policy.name, extras)
+
         if isinstance(resource, IAMRole):
             self._check_policy_document(result, logical_id, resource.Properties.AssumeRolePolicyDocument, None, extras)
-        elif isinstance(resource, KMSKey):
-            self._check_policy_document(result, logical_id, resource.Properties.KeyPolicy, None, extras)
```

Comment:
why is this not needed anymore?

Reply:
Put that in the commit message, I wasn't clear enough, my bad 😞 
0.17.0 of pycfmodel -> https://github.com/Skyscanner/pycfmodel/releases/tag/0.17.0
allows us to retrieve KMS Key policies by using the parent property of `policy_documents`